api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceService.java

```diff
@@ -45,6 +45,8 @@
 
   DbWorkspace saveWithLastModified(DbWorkspace workspace);
 
+  void deleteWorkspace(DbWorkspace dbWorkspace);
```

Comment:
If it's not too tricky, I'd probably return some kind of status for whether the workspace was successfully deleted, wasn't there in the first place, or was otherwise not deletable. Might want to capture this on both our side and the firecloud side.

Reply:
my preference is to handle failures through exceptions

Reply:
But if you're deleting 1000 workspaces and 3 of them fail, an exception at the outer level isn't that helpful. Since we're in an eventually-consistent-ish system, I think you have to consider partial success cases (especially if it's not all in one transaction).